### PR TITLE
Chore: remove `react-popper-tooltip` in favour of `@floating-ui/react`

### LINKED ISSUE #59350: Hover panels overlap useful content in mobile browsers
The panels that are supposed to appear on mouse hover in desktop browsers, when on mobile, appear on tap.

image

image

After appearing, they overlap controls underneath and do not allow to use them. Thus, it is very tricky to tap on the date picker or magnifying glass on the first screenshot and it is an annoyance on the second screenshot.

**Expected behaviour:**
Disable hover panels on mobile version, as there is no hover event.

**Environment**:
- Grafana version: Cloud
- User OS & Browser: iOS 16.1.1 & Chrome 107.0.5304.101
- Device: iPhone SE 2020
- View port size: 375px × 667px

Please note that the issue is not reproducible in Desktop Chrome with dev tools, the hover panel does not appear on tap.

### LINKED ISSUE #78615: ToggleTip display issue on v10.2.2
### What happened?

I'm developing my own grafana plugin. When I upgraded my Grafana from 10.0.2 to 10.2.2 and upgraded all Grafana dependencies versions of my plugin from 10.0.2 to 10.2.2, I found there was a display issue on the ToggleTip ui component. 

- On the display page, there is a small margin between the button and the tip.
image

- On the editing page, there is a big margin between the button and the tip.
image

Here is my code: 
```
interface Props {
  explain: string
}

const Explain: React.FC<Props> = (props) =>{
  let html =
    <div className={"panel-info-content markdown-html"}>
      <div dangerouslySetInnerHTML={{ __html: props.explain }} />
    </div>

  return <Toggletip content={html} closeButton={true} placement="right-start">
    <IconButton name="file-alt" size="md" variant="secondary" tooltip={"Explain"}/>
  </Toggletip>
}
```

### What did you expect to happen?

It shouldn't have a margin between the button and the tip.

### Did this work before?

In the version 10.0.2, it works fine.

### How do we reproduce it?

1.
2.
3.

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: 10.2.2
OS: Both Linux and MaxOS
Browser: Chrome

### Grafana platform?

Docker

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- my christmas gift 🎅 removes `react-popper-tooltip` and replaces all uses with `@floating-ui/react`
- there are 5 uses in total, listed below roughly in order of importance:
  - `Tooltip`
  - `Dropdown`
  - `Toggletip`
    - | before | after |
      | --- | --- |
      | image | image |
  - `OperationInfoButton`
  - `NestedFolderPicker` 
- reinstates `restoreFocus` in `TimeRangePicker` and `AppChromeMenu` 🥳 
- also stops setting `aria-describedby` for a tooltip **if the tooltip text matches the existing `aria-label`**
  - this prevents double announcing on `IconButton`s

**Why do we need this feature?**

- `@floating-ui/react` is the next evolution of popper and has (imo) a much nicer/cleaner api
- `@floating-ui` makes a distinction between `focus` and `focus-visible`
  - this means we can now properly use `restoreFocus` without it showing tooltips again
  - as an example, consider the `TimeRangePicker`
    - we had to remove `restoreFocus` because it was too annoying having the tooltip show again when you selected a new time range or just closed the overlay. this is good for normal usability but bad for a11y!
    - with `@floating-ui`, the tooltip is only shown again **if you use the keyboard** to dismiss the overlay
- i think it also fixes https://github.com/grafana/grafana/issues/59350
  - at least i can't repro as described in the issue on my phone
  - i DO see a tooltip flash which isn't ideal, but it's definitely much better than the issue describes
- eventually we can also completely remove `react-popper` and replace with `@floating-ui` as well, but i wanted to keep this initial PR smaller so just focused on `react-popper-tooltip`

**Who is this feature for?**

- everyone!

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/78615
Fixes https://github.com/grafana/grafana/issues/59350

**Special notes for your reviewer:**

i haven't really made any improvements here. going through these components, i think there are now several we can make with floating-ui which will make things nicer. couple of main ones:
1. `OperationInfoButton` should really just use `Toggletip` imo
1. we can use `FloatingFocusManager` to close toggletips when you tab out of them 💡 

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Toggletip/Toggletip.tsx
Comment (academo) [inline packages/grafana-ui/src/components/Toggletip/Toggletip.tsx:104]: Fuzzy memory. Does this affect somehow the fix for https://github.com/grafana/grafana/issues/78615 ?

Comment (ashharrison90) [reply packages/grafana-ui/src/components/Toggletip/Toggletip.tsx:104]: nope - this PR still fixes that issue

this is on the branch with the test plugin you provided in https://raintank-corp.slack.com/archives/C025WTVRBSN/p1702636168315079:

image

Comment (academo) [reply packages/grafana-ui/src/components/Toggletip/Toggletip.tsx:104]: Thanks for double checking. I somehow had in my mind that "fixed" was causing this issue but I was wrong. :)

**Comment by joshhunt** [comment]
I've spotted one regression/change in behaviour that is pretty minor, but I think could be an annoying papercut.

When opening the time picker, any movement on the original button triggers the tooltip to appear again. When you move your mouse down to the time picker dropdown, the tooltip appears which is a bit annoying.

image

On my instance without this change, the tooltip doesn't appear until the mouse is re-entered on the button. When you move your mouse down to the time picker dropdown, the tooltip _doesnt_ appear.

**Comment by ashharrison90** [comment]
great spot @joshhunt 🥳 

there's an option for that! https://floating-ui.com/docs/useHover#move (can you tell i love this api)

i don't _think_ it will affect anything else, since like you said it's just using the old behaviour. i'll make the change and update the ephemeral instance 👍

**Review by joshhunt — APPROVED** [review]
Verified that the previous timepicker issue is resolved. Done a smoke test around and everything seems okay. 

Happy to merge this in if other codeowners don't have objections.

File: packages/grafana-ui/src/components/Toggletip/Toggletip.story.tsx
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/Toggletip/Toggletip.story.tsx:18]: Why did we add this to the controls? 🤔

**Review by JoaoSilvaGrafana — APPROVED** [review]
It actually deals with moving the toggletip when you resize the window 🥲 

https://github.com/grafana/grafana/assets/100691367/59ef41b5-72a2-49e7-a08c-f72bd28cf590

This looks great, one small question, but approved anyway 👏

**Comment by ashharrison90** [comment]
@JoaoSilvaGrafana 

> It actually deals with moving the toggletip when you resize the window 🥲

i know, it's beautiful 🥲 

> Why did we add this to the controls? 🤔

ah it was to be consistent with `onOpen`. either we can remove both from the controls, or have them both in. any preference?

think we should also change the type for `onClose` to be the same as `onOpen`.

**Comment by JoaoSilvaGrafana** [comment]
> ah it was to be consistent with onOpen. either we can remove both from the controls, or have them both in. any preference?
> think we should also change the type for onClose to be the same as onOpen.

I'm ok with either of these options. We can make them the same type so at least `onClose` is not editable as a control as well?

**Review by JoaoSilvaGrafana — APPROVED** [review]
Nice

**Pull request merged**
